src/frontend/src/pages/InstanceDetail.tsx

```diff
@@ -0,0 +1,28 @@
+import { LoadingOverlay } from '@mantine/core';
+
+import ClientError from './errors/ClientError';
+import ServerError from './errors/ServerError';
+
+export default function InstanceDetail({
+  status,
+  loading,
+  children
+}: {
+  status: number;
+  loading: boolean;
+  children: React.ReactNode;
+}) {
+  if (loading) {
+    return <LoadingOverlay />;
+  }
+
+  if (status >= 500) {
+    return <ServerError status={status} />;
+  }
+
+  if (status >= 400) {
+    return <ClientError status={status} />;
+  }
```

Comment:
Question: What is the error state if the user has viewing perms but can not do a change action? Are they immediately thrown off the page?

Reply:
I do not think that this update will have any issues here - the permissions checks should happen separately. Certainly will keep an eye out for it